### ISSUE TITLE
recent linux kernel changes break ARM cpuinfo parser

### ISSUE THREAD

**Issue by danielbprice** [joyent/libuv#812]
I am running node under a recent (3.9.3) linux kernel on ARM.  I was previously using node 0.8.22, and was seeing crashes creating the "model" string when parsing /proc/cpuinfo.  Node 0.10.8 appears to instead categorize my CPUs as "unknown".  So then the question is: why is the code failing to find my CPU model?

This is because the format of /proc/cpuinfo has apparently changed incompatibly in recent kernels.  The old format was:

```
Processor       : ARMv6-compatible processor rev 7 (v6l)
```

The new format is (on a slightly different box)

```
processor       : 0
model name      : ARMv7 Processor rev 2 (v7l)
...
```

So in other words this has become much more like x86 and is multi-processor capable.  I tracked the linux change down to this:

https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=82b5df7bf27ca6cf642499bc3bcae1c93b9ac4ba

Apparently this work is in advance of coming "big.LITTLE" ARM CPUs, and it looks like it's in 3.8 series kernels.

So as best I can tell, the old check for "Processor:" will no longer work, and the code will have to parse for "model name:" instead.  Perhaps the code could be merged such that it looks for model name: and failing that falls back to "Processor:"?  That code might then work on x86 and ARM.

Hopefully I've worked this out correctly and this is not a duplicate issue.  Apologies otherwise.

**Comment by bnoordhuis** [comment]
Thanks for the report. Does bnoordhuis/libuv@4f92b5c fix it?

**Comment by danielbprice** [comment]
Yes, it does seem to work!  Thank-you.

Also, I did notice that this code is making unchecked calls to strndup()-- opening up the possibility that, if the string was not able to be dup'd for whatever reason, that this would propagate NULL all the way back into nodejs; maybe that's a different problem, or you don't care.  Just thought I would mention it.

I do get this warning when building:

```
../deps/uv/src/unix/linux-core.c: In function 'read_models':
../deps/uv/src/unix/linux-core.c:423:16: warning: variable 'speed_idx' set but not used [-Wunused-but-set-variable]
```

Which is related to the CPU frequency code.

[edited to reflect my evolving understanding of this code; disregard my drivel about bogomips]

**Comment by bnoordhuis** [comment]
Thanks, landed in 92c72f5 with a fix-up to suppress that unused variable warning. I've added OOM handling in b93cf8b.
